src/resource/archive/ArchiveManager.cpp

```diff
@@ -64,10 +64,41 @@ bool ArchiveManager::HasFile(uint64_t hash) {
     return mFileToArchive.count(hash) > 0;
 }
 
-std::shared_ptr<std::vector<std::string>> ArchiveManager::ListFiles(const std::string& filter) {
+std::shared_ptr<std::vector<std::string>> ArchiveManager::ListFiles() {
+    return ListFiles({}, {});
+}
+
+std::shared_ptr<std::vector<std::string>> ArchiveManager::ListFiles(const std::string& searchMask) {
+    return ListFiles({ searchMask }, {});
+}
```

Comment:
do we need both of these or could the line in `ArchiveManager.h` just change? 
```diff
- std::shared_ptr<std::vector<std::string>> ListFiles(const std::string& searchMask);
+ std::shared_ptr<std::vector<std::string>> ListFiles(const std::string& searchMask = "");
```

Reply:
Doing so would require me to change how the main function checks for non-existent filters, as a list initialized with `""` I believe doesn't count for `List::empty()`

Reply:
i think i'm just imagining times when a port might do something like
```cpp
std::string mySearchMask = "";
if (customLogic) {
  mySearchMask += customLogicPath;
}
ListFiles(mySearchMask);
```
so handling the case where `customLogic` is false there the same as someone just calling `ListFiles()` seems reasonable

Reply:
Eh, I guess it's easy enough to change the way the searchMask function passes the searchMask to the main one so that it doesn't add a size 0 string to the list before passing it.